draft-ietf-quic-http.md

```diff
@@ -1680,24 +1692,172 @@ extension is disabled if the setting is omitted.
-
-TODO:  This is going to be a big import, probably worthy of its own PR.
-
-When HTTP Alternative Services is used for discovery for HTTP/3 endpoints, the
-security considerations of {{!ALTSVC}} also apply.
-
-## Traffic Analysis
+with TLS.  However, many of the considerations from Section 10 of {{?HTTP2}}
+apply to [QUIC-TRANSPORT] and are discussed in that document.
+
+## Server Authority
+
+HTTP/3 relies on the HTTP definition of authority. The security considerations
+of establishing authority are discussed in Section 11.1 of {{!SEMANTICS}}.
+
+## Cross-Protocol Attacks
+
+The use of ALPN in the TLS and QUIC handshakes establishes the target
+application protocol before application-layer bytes are processed.  Because all
+QUIC packets are encrypted, it is difficult for an attacker to control the
+plaintext bytes of an HTTP/3 connection which could be used in a cross-protocol
+attack on a plaintext protocol.
+
+## Intermediary Encapsulation Attacks
+
+The HTTP/3 field encoding allows the expression of names that are not valid
+field names in the syntax used by HTTP (Section 4.3 of {{!SEMANTICS}}). Requests
+or responses containing invalid field names MUST be treated as malformed
+({{malformed}}).  An intermediary therefore cannot translate an HTTP/3 request
+or response containing an invalid field name into an HTTP/1.1 message.
+
+Similarly, HTTP/3 allows field values that are not valid. While most of the
+values that can be encoded will not alter field parsing, carriage return (CR,
+ASCII 0xd), line feed (LF, ASCII 0xa), and the zero character (NUL, ASCII 0x0)
+might be exploited by an attacker if they are translated verbatim. Any request
+or response that contains a character not permitted in a field value MUST be
+treated as malformed ({{malformed}}).  Valid characters are defined by the
+"field-content" ABNF rule in Section 4.4 of {{!SEMANTICS}}.
+
+## Cacheability of Pushed Responses
+
+Pushed responses do not have an explicit request from the client; the request is
+provided by the server in the PUSH_PROMISE frame.
+
+Caching responses that are pushed is possible based on the guidance provided by
+the origin server in the Cache-Control header field. However, this can cause
+issues if a single server hosts more than one tenant.  For example, a server
+might offer multiple users each a small portion of its URI space.
+
+Where multiple tenants share space on the same server, that server MUST ensure
+that tenants are not able to push representations of resources that they do not
+have authority over.  Failure to enforce this would allow a tenant to provide a
+representation that would be served out of cache, overriding the actual
+representation that the authoritative tenant provides.
+
+Pushed responses for which an origin server is not authoritative (see
+{{connection-reuse}}) MUST NOT be used or cached.
+
+## Denial-of-Service Considerations
+
+An HTTP/3 connection can demand a greater commitment of resources to operate
```

Comment:
I'm not sure I know why this would be true, particularly for HTTP/2 vs HTTP/3?

Reply:
Some implementations still experience higher resource usage for HTTP/3 for various reasons (lack of hardware offload, smaller encryption units, UDP less optimized than TCP, etc.).  While I hope that some of those things will change over time and the rest will matter less, I think it's still a generally true statement that HTTP/3 **can** demand greater resources.